packages/target-tools/src/telemetryProvider.spec.js

```diff
@@ -111,56 +119,68 @@ describe("TelemetryProvider", () => {
       })
     );
   });
-
-  it("assigns local execution mode", () => {
+  it("assigns edge execution mode for server side", () => {
     const provider = TelemetryProvider();
-
     provider.addDeliveryRequestEntry(
       TARGET_REQUEST,
       TARGET_TELEMETRY_ENTRY,
-      STATUS_OK,
-      DECISIONING_METHOD.ON_DEVICE
+      DELIVERY_RESPONSE,
+      DECISIONING_METHOD.SERVER_SIDE
     );
+    const entries = provider.getAndClearEntries();
+    expect(entries[0].mode).toEqual(EXECUTION_MODE.EDGE);
+  });
+
+  it("assigns local execution mode for on device", () => {
+    const provider = TelemetryProvider();
+    const DELIVERY_RESPONSE_ON_DEVICE = Object.assign({}, DELIVERY_RESPONSE, {
+      edgeHost: null
+    });
+
     provider.addDeliveryRequestEntry(
       TARGET_REQUEST,
       TARGET_TELEMETRY_ENTRY,
-      STATUS_OK,
-      DECISIONING_METHOD.HYBRID
+      DELIVERY_RESPONSE_ON_DEVICE,
+      DECISIONING_METHOD.ON_DEVICE
     );
-
     const entries = provider.getAndClearEntries();
-
     expect(entries[0].mode).toEqual(EXECUTION_MODE.LOCAL);
-    expect(entries[1].mode).toEqual(EXECUTION_MODE.LOCAL);
   });
 
-  it("assigns edge execution mode", () => {
+  it("assigns edge execution mode for hybrid when execution in not all local", () => {
     const provider = TelemetryProvider();
 
+    const DELIVERY_RESPONSE_HYBRID_1 = Object.assign({}, DELIVERY_RESPONSE, {
```

Comment:
Don't need to create a new DELIVERY_RESPONSE here since the same edgeHost value is already set on the base DELIVERY_RESPONSE.